### DISCUSSION TITLE
Rules matching speed

### DISCUSSION THREAD

**Discussion by chu121su12** [unocss/unocss#188] [Ideas]
Do you think unocss can benefit from matching combined rules? Considering classes need to be matched to some rules, I am reminded to the improvements made to symfony router by combining route definition into a single regex (see references).

References: 
- https://symfony.com/blog/new-in-symfony-4-1-fastest-php-router
- https://www.npopov.com/2014/02/18/Fast-request-routing-using-regular-expressions.html
- https://medium.com/@nicolas.grekas/making-symfonys-router-77-7x-faster-1-2-958e3754f0e1
- https://medium.com/@nicolas.grekas/making-symfony-router-lightning-fast-2-2-19281dcd245b

**Comment by antfu** [comment]
Yes, that's indeed some optimization that could be done in the presets or userland. But giving the fact that the engine is blazing fast, and we do a good cache for matched strings I don't see this micro regex cost could affect the overall performance much - at least for now.
